Добавил api-сервисы

CaplexW opened the pull request:
Добавил сервисы для обработки данных о:
- Пользователе
- Фильмах
Дальнейшие методы будут добавляться по мере разработки фич.

AleksandrMenshchikov committed:
Создать страницу Home

Commit:
Добавил сервисы для обработки даных о:
- Пользователе
- Фильмах
Дальнейшие методы будут добавляться по мере разработки фич.

Commit:
Добавил сервисы для обработки даных о:
- Пользователе
- Фильмах
Дальнейшие методы будут добавляться по мере разработки фич.

AleksandrMenshchikov commented on src/services/film.service.ts:
```diff
@@ -0,0 +1,30 @@
+
+const filmService = {
+  async getTopRatedSeries(): Promise<Film[] | undefined>  {
+    const LIST_NAME = 'top_rated_series_250';
```
Александр, данный сервис реализован в redux/api/filmApi.ts

CaplexW replied:
На момент когда я его писал, ещё не было redux/api/filmApi.ts xD
C createApi раньше не работал. Разберусь в нем на днях и приведу все к общему знаменателю.

AleksandrMenshchikov commented on src/services/user.service.ts:
```diff
@@ -0,0 +1,28 @@
+
+const userService: UserService = {
+  getUserInfo(id: string): Promise<UserData> {
+    return new Promise((res, rej) => {
```
Думаю можно упростить примерно так

class UserLocalStorage {
    private static USER = 'user';

    static getUser() {
      localStorage.getItem(UserLocalStorage.USER);
    }
    
    static setUser(user) {
      localStorage.setItem(UserLocalStorage.USER, user);
    }
    
    static removeUser() {
      localStorage.removeItem(UserLocalStorage.USER);
    }
  }

CaplexW replied:
Идея была в том, чтобы сделать асинхронный запрос к "серверу", а не просто инкапсулировать логику.

AleksandrMenshchikov commented on src/types/Film.ts:
```diff
@@ -0,0 +1,43 @@
+export type Film = {
+  id: string;
```
по Типам сложно комментировать, они по ходу работы будут меняться, все поля вероятно не нужны будут.

CaplexW replied:
Согласен. Не понятно, что будет, но убирать проще чем добавлять.

AleksandrMenshchikov committed:
Перенести Wrapper в components

CaplexW committed:
backup save

CaplexW committed:
backup save

CaplexW committed:
Backup save

CaplexW committed:
Backup save

AleksandrMenshchikov commented on src/utils/generateUniqId.ts:
```diff
@@ -0,0 +1,3 @@
+export default function generateUniqId(): string {
+  return `${Date.now() - Number(Math.random().toString(36))}`;
+}
```
Функция возвращает NaN

CaplexW replied:
Да, она не была доделана на тот момент. В ПР попало то, что я не планировал.

Commit:
backup save

Commit:
backup save

Commit:
backup save

Commit:
backup save

Commit:
backup save

Commit:
backup save

Commit:
backup save

Commit:
backup save

CaplexW committed:
Подтянул свежий dev

CaplexW committed:
Добавлен 'бэкенд' и сервисы для работы с ним.
1.В fakeServer реализованы методы:
 - signUp
 - singIn
 - updateUser
 - getUserById
 - getFilmList
2. Добавлен authService для работы с авторизацией.
3. Добавлен userService для работы с данными пользователей.
4. Добавлена утилита handleError для обработки ошибок. Её функционал на данный момент символический и будет дорабатываться по мере разработки.

CaplexW committed:
Форматирование prettier
